helm-charts/ironic/ironic-config/dnsmasq.conf.j2

```diff
@@ -31,6 +31,13 @@ dhcp-boot=tag:efi,tag:!ipxe,snponly.efi
 dhcp-boot=/undionly.kpxe,{{ env.IRONIC_IP }}
 {% endif %}
 
+{% for mac in (env.DHCP_HOSTS | default('#') | replace('-', ':') | replace('\\,', ',')).split(',') -%}
```

Comment:
This block result in dnsmasq failed to start because of invalid syntax if DHCP_HOSTS is not specified.
```
dhcp-host=#
```

I think we need to check if DHCP_HOSTS is specified first by doing this.
```
{%- if env.DHCP_HOSTS | length %}
{% for mac in (env.DHCP_HOSTS | replace('-', ':') | replace('\\,', ',')).split(',') -%}
{%- if mac | length %}
dhcp-host={{ mac }}
{% endif %}
{% endfor %}
dhcp-ignore=tag:!known
{% endif %}
```

Reply:
I think dhcp-host=# need to be quoted. I will try that out i.e. dhcp-host="#", but yes that is needed. I thought I had done that. I must have dropped that change somewhere along the way. Will get the change up in a bit.